pkgs/development/compilers/ghc/head.nix

```diff
@@ -12,7 +12,7 @@ let
 
   commonBuildInputs = [ ghc perl autoconf automake happy alex python3 ];
 
-  version = "8.1.20170106";
+  version = "2017-01-06";
```

Comment:
Please feel free to suggest better version numbers for the ghc and ghcjs expressions in a new PR. Then we can discuss that choice and reach a consensus. Meanwhile, I reverted your changes in a1b6c5f5d6f22088dc017552c94d133128059ad8.